### PR TITLE
Add dynamic dispatch to AIX and clang for Power

### PR DESCRIPTION
**Pull request by ChipKerchner**
Known issue (to be done as a separate PR):  compiling with gcc/xlf in AIX.
Works with Linux - gcc & clang,  AIX - gcc/gfortran, xlc/xlf.

This adds dynamic dispatch to AIX.  It also adds dynamic dispatch for clang/LLVM for Linux.  All for PowerPC

### CODE REVIEW

File: driver/others/dynamic_power.c [hunk 1]
```diff
@@ -36,13 +36,38 @@ char *gotoblas_corename(void) {
 	return corename[0];
 }
 
-#if defined(__clang__)
-static int __builtin_cpu_supports(char* arg) 
+#define CPU_UNKNOWN  0
+#define CPU_POWER5   5
+#define CPU_POWER6   6
+#define CPU_POWER7   7
```
Comment (martin-frbg) [inline driver/others/dynamic_power.c]: POWER7 currently does not exist as an actual, unique target - it is an alias for POWER6. Do you plan to change that ?

Comment (ChipKerchner) [reply driver/others/dynamic_power.c]: Fixed

**Review by martin-frbg — COMMENTED** [review]
thanks - I guess arch==power7 would still be valid and return CPU_POWER6 though?

**Comment by ChipKerchner** [comment]
Fixed.  We would like these changes in the next release (Nov 5th)

**Pull request merged**